service/src/main/java/org/cbioportal/service/impl/UuidDataAccessTokenServiceImpl.java

```diff
@@ -0,0 +1,203 @@
+    @Override
+    public DataAccessToken getDataAccessToken(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        DataAccessToken newestDataAccessToken = getNewestDataAccessToken(allDataAccessTokens);
+        return newestDataAccessToken;
+    }
+
+    @Override
+    public DataAccessToken getDataAccessTokenInfo(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        return dataAccessToken;
+    }
+
+    @Override
+    public void revokeAllDataAccessTokens(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        for (DataAccessToken dataAccessToken : allDataAccessTokens) {
+            dataAccessTokenRepository.removeDataAccessToken(dataAccessToken.getToken());
+        } 
+    }
+
+    @Override
+    public void revokeDataAccessToken(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        dataAccessTokenRepository.removeDataAccessToken(dataAccessToken.getToken());
+    }
+    
+    @Override
+    public String getUsername(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        return dataAccessToken.getUsername();
+    }
+    
+    @Override
+    public Date getExpiration(String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenRepository.getDataAccessToken(token);
+        return dataAccessToken.getExpiration();
+    }
+   
+    @Override
+    public Boolean isValid(String dataAccessToken) {
+        DataAccessToken storedDataAccessToken = null;
+        try {
+            storedDataAccessToken = dataAccessTokenRepository.getDataAccessToken(dataAccessToken);
+        } catch (Exception e) {
+            log.error("Error retrieving data access token, " + dataAccessToken + " from token store");
+            return Boolean.FALSE;
+        }
+        Calendar calendar = Calendar.getInstance();
+        Date currentDate = calendar.getTime();
+        if (storedDataAccessToken.getExpiration().before(currentDate)) {
+            return Boolean.FALSE;
+        } 
+        return Boolean.TRUE;
+    }
+
+    private int getNumberOfTokensForUsername(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        return allDataAccessTokens.size();
+    }
+
+    // revokes oldest token in token management system for a user
+    private void revokeOldestDataAccessTokenForUsername(String username) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenRepository.getAllDataAccessTokensForUsername(username);
+        DataAccessToken oldestDataAccessToken = getOldestDataAccessToken(allDataAccessTokens);
```

Comment:
finding the oldest token can be done within mysql using an ORDER BY clause and an index in the table for the expiration date. This will be faster than linearly scanning the tokens below.